tools/omnissm/pkg/manager/manager.go

```diff
@@ -57,27 +86,40 @@ func New(config *Config) *Manager {
 	for _, t := range config.ResourceTags {
 		m.resourceTags[t] = struct{}{}
 	}
+	if config.QueueName != "" {
+		m.queue = NewQueue(config.QueueName, config.Config)
+	}
 	return m
 }
 
 func (m *Manager) Register(doc *identity.Document) (*store.RegistrationEntry, error) {
-	resp, err := m.SSMAPI.CreateActivation(&ssm.CreateActivationInput{
-		DefaultInstanceName: aws.String(doc.Name()),
-		IamRole:             aws.String(m.ssmInstanceRole),
-		Description:         aws.String(doc.Name()),
+	var entry *store.RegistrationEntry
+	err := retry(m.maxRetries, func() error {
+		resp, err := m.SSMAPI.CreateActivation(&ssm.CreateActivationInput{
+			DefaultInstanceName: aws.String(doc.Name()),
+			IamRole:             aws.String(m.ssmInstanceRole),
+			Description:         aws.String(doc.Name()),
+		})
+		if err != nil {
+			return err
+		}
+		entry = &store.RegistrationEntry{
+			Id:             identity.Hash(doc.Name()),
+			ActivationId:   *resp.ActivationId,
+			ActivationCode: *resp.ActivationCode,
+		}
+		return m.Put(entry)
```

Comment:
i think dynamodb api call should be out of closure, retries on it are a sign of capacity problems for most cases, and should be a sign of adjustment needed, or at least log output

Reply:
the idea was that dynamodb would retry internal to the call since only the ssm had the default max retries turned off, but 1) i see that as not being particular clear 2) i neglected to put check here that would do the desired control flow given that.  so with that said, i think i can implement it the way you suggested and ensure that log output appropriately reflects the potential capacity issues as well.